### PR TITLE
Add Travis config file

### PR DESCRIPTION
**Pull request by ararslan**
This isn't really a "package," per se, so rather than using the default Travis test script, we'll just try generating everything then make sure it did something.

A stage for building the documentation has been included but is currently commented out. That will be uncommented later once everything is up and running smoothly.

Something we may want to consider is having Travis run on a cron and automatically open pull requests if there are new changes for e.g. the Services submodule in AWSCore.

### CODE REVIEW

**Comment by ararslan** [comment]
GitHub API rate limited. 😑

**Comment by ararslan** [comment]
We're getting a connection refused error when trying to POST to the chosen port on localhost: https://travis-ci.com/JuliaCloud/AWSAPI.jl/builds/99850131#L505. I was getting this locally as well but I fixed it by pointing Node to the correct directory for packages. Unfortunately that does not seem to help Travis.

File: src/AWSMetadata.jl [hunk 1]
```diff
@@ -20,6 +20,13 @@ using Retry
 
 github_headers = ["User-Agent" => "https://github.com/JuliaCloud/AWSCore.jl/blob/master/src/AWSMetadata.jl"]
```
Comment (omus) [inline src/AWSMetadata.jl]: I doubt this matters but it should probably be changed:
```suggestion
 github_headers = ["User-Agent" => "https://github.com/JuliaCloud/AWSAPI.jl/blob/master/src/AWSMetadata.jl"]
```

**Comment by rofinn** [comment]
Might be helpful to add something like:

```
console.log('Listening on port $port...');
```

After starting the http server to help debug issues? Apart from that, I'm assuming the npm output is coming from the server, so maybe extend the sleep time?

**Comment by ararslan** [comment]
I just realized that this is making like 1000 GitHub API requests for individual files all from the same repo. It seems to me it could just do a shallow clone of the repo. There's no way this is faster...

**Pull request merged**